Encourage deferring triage to NV access for features and product decisions (#17315)
nvaccess/nvda: projectDocs/issues/triage.md

New features and enhancements should be [well defined](#new-features-1) before applying the `triaged` label.

> Once a bug has clear steps to reproduce and is well documented, the `triaged` label can be applied.

Reviewer: This line (and the one following) suggests that community members are fine to triage and prioritize bugs. Is that correct?

Reviewer 2: @XLTechie That's right. Our rubric is:
- Well formed bugs following the issue template, which can easily be understood and isn't missing any important debug information
- Can clearly be reproduced
- Can clearly be prioritised
- Doesn't require a controversial solution

A `triaged` issue should also have a [priority label](#priority).

> Community members should avoid adding a `triaged` label to feature requests or other product decisions, deferring to NV Access.

Reviewer 3: I do not understand what "or other product decisions" is referring to. Any clarification?

> Community members should avoid adding a `triaged` label to feature requests or other product decisions, deferring to NV Access.

Reviewer 2: @CyrilleB79 @XLTechie How about this?

```suggestion
Community members should avoid adding a `triaged` label to feature requests or other decisions potentially involving significant or controversial changes to NVDA features or functionality, which may require community input or approval from NV Access.
```

Reviewer 4: Should clarify "may require community input" to "may require more extensive community consultation"?

(Just thinking any member of the community reading a given issue and wanting to triage something could arguably be providing "community input")

> The label should also be avoided for other issues that are controversial, or the priority is unclear, such as bug fixes with an unclear solution.

Reviewer 5: ```suggestion
The label should also be avoided for other issues that are controversial, or where the priority is unclear, such as bug fixes with an unclear solution.
```

> For changes where a product decision from NV Access is required before applying the `triaged` label, the label `blocked/needs-product-decision` should be used.

Author: ```suggestion
For changes where a product decision from NV Access is required before applying the `triaged` label, the label `blocked/needs-product-decision` should be used.
Things to consider before applying the `triaged` label:

- Well formed bugs following the issue template, which can easily be understood and isn't missing any important debug information
- Can clearly be reproduced
- Can clearly be prioritised
- Doesn't require a controversial solution
```

Once a bug has clear steps to reproduce and is well documented, the `triaged` label can be applied.
A `triaged` issue should also have a [priority label](#priority).

Community members should avoid adding a `triaged` label to feature requests or other decisions potentially involving significant or controversial changes to NVDA features or functionality, which may require community input or approval from NV Access.
The label should also be avoided for other issues that are controversial, or where the priority is unclear, such as bug fixes with an unclear solution.
For changes where a product decision from NV Access is required before applying the `triaged` label, the label `blocked/needs-product-decision` should be used.
Community members should apply the `triaged` where an issue is:

* A well formed bug report following the issue template
* Which can easily be understood
* Isn't missing any important debug information
* Can clearly be reproduced
* Can clearly be prioritised
* Doesn't require a controversial solution

If an issue has been checked by NV Access, and needs further triage, the `needs-triage` label will be applied.

> Please notify NV Access when you believe the issue is ready for the `triaged` label.

Reviewer 6: ```suggestion
Please notify NV Access when you believe the issue is ready for the `triaged` label.
You can do it by applying the NeedsTriage label, or cc @nvaccess in the issue, or mailing to info@nvaccess.org
```
In short, I don't know how this notification should be done. Can you clarify this please?

New features and enhancements should be [well defined](#new-features-1) before applying the `triaged` label.
Once a bug has clear steps to reproduce and is well documented, the `triaged` label can be applied.
A `triaged` issue should also have a [priority label](#priority).

> Community members should avoid adding a `triaged` label to feature requests or other decisions potentially involving significant or controversial changes to NVDA features or functionality, which may require community input or approval from NV Access.

Author: I think "where" or just removing "other" makes more sense here

```suggestion
Community members should avoid adding a `triaged` label to feature requests or decisions potentially involving significant or controversial changes to NVDA features or functionality, which may require community input or approval from NV Access.
```

> The label should also be avoided for other issues that are controversial, or where the priority is unclear, such as bug fixes with an unclear solution.

Reviewer 7: I think we need to make the difference between the issue reported, and the bug fix submited as a pr. Bug fixes can be controversial from an UX perspective (e.g. a pr that introduces a degradation in user experience or other significant risks).  On the other hand,  a reported issue can be unclear or not reproducible. But it is difficult in my view to define „controversial“ for a reported bug. I would rather replace „controversial“ by „unclear community support“. This is quite easy to distinguish, e.g. many comments where people exchange pro and contra arguments back and forth, and there is no clear conclusion on how a solution could look like.

Author: After internal discussion we'd prefer to keep the current wording. Unclear community support requires some back and forth discussion. However, we'd want to ensure controversial issues aren't triaged before that conversation has a chance to occur. 
When in doubt, if its unclear if the solution or issue seems controversial, we'd simply encourage not adding the triaged label.

For changes where a product decision from NV Access is required before applying the `triaged` label, the label `blocked/needs-product-decision` should be used.

> Community members should apply the `triaged` where an issue is:

Reviewer 5: ```suggestion
Community members should apply the `triaged` label where an issue is:
```

> * A well formed bug report following the issue template

Reviewer: If you take my suggestion about the introduction:

```suggestion
```

> * Which can easily be understood

Reviewer: If you take my suggestion above re the introduction, remove the "Which" here. If you don't, "which" should be merged into the parent item, since each of the following could also start with "which".

```suggestion
* Can easily be understood
```

* Isn't missing any important debug information
* Can clearly be reproduced
* Can clearly be prioritised

> * Doesn't require a controversial solution

Author: Removing it would create a more vague set of expectations. Could you suggest what's missing or not objective? We tend to follow this process more-or-less when triaging issues as NV Access. In our experience its quite clear and objective if something can be reproduced, has attached debug logs, and is prioritised in our label system.

> * Doesn't require a controversial solution

Author: see https://github.com/nvaccess/nvda/pull/17315#discussion_r1839419962

> If an issue has been checked by NV Access, and needs further triage, the `needs-triage` label will be applied.

Author: We apply those additionally if needed, but we want a label for issues we have reviewed, and determined it is not in a triaged state.

Please notify NV Access when you believe the issue is ready for the `triaged` label.

> You can do this by tagging @nvaccess/developers or mailing info@nvaccess.org.

Reviewer 5: ```suggestion
You can do this by tagging @nvaccess/developers or emailing info@nvaccess.org.
```

> You can do this by tagging @nvaccess/developers or emailing info@nvaccess.org.

Reviewer: ```suggestion
You can do this by tagging `@nvaccess/developers` or emailing `info@nvaccess.org`.
```

Author: ```suggestion
You can do this by tagging `@nvaccess/developers` or emailing <info@nvaccess.org>.
```

> A `triaged` issue that requires a complex fix may require advice from NV Access, such as a project plan, before implementation is started.

Author: This is more relevant to those planning on turning a triaged issue into a PR.
The idea is that if you see the "triaged" label, while it is ready to be worked on, it might be best to write up a proposed project plan before you start coding.
